test/storage.test.js

```diff
@@ -287,4 +288,158 @@ describe('storage.js', () => {
+                assert.ok(e instanceof GenericError);
+                assert.equal(e.message, "GET 'https://example.com/photo/elephant.png' failed with status 404");
+                threw = true;
+            }
+            assert.ok(threw);
+        });
+
+        it('should not download a file in worker test mode', async () => {
+            process.env.WORKER_TEST_MODE = true;
+
+            const params = {
+                watermarkContent: 'file.png'
+            };
+            const inDirectory = '/in';
+
+            mockFs({ '/in/file.png': 'yo' });
+            const watermark = await getWatermark(params, inDirectory);
+
+            assert.equal(watermark.name, 'watermark.png'); // in this case source name is actual file path
+            assert.equal(watermark.path, '/in/watermark.png');
+        });
+
+        it('should fail to download because path ends with /..', async () => {
+            process.env.WORKER_TEST_MODE = true;
+            const params = {
+                watermarkContent: 'file.jpg/..'
+            };
+            const inDirectory = '/in';
+
+            let threw = false;
+            try {
+                await getWatermark(params, inDirectory);
+            } catch (e) {
+                assert.equal(e.message, 'Invalid or missing local file file.jpg/..');
+                threw = true;
+            }
+            assert.ok(threw);
+        });
+
+        it('should fail when path contains traversal notation in worker test mode', async () => {
+            process.env.WORKER_TEST_MODE = true;
+            const params = {
+                watermarkContent: 'file/../../../../evilcode/elephant.jpg'
+            };
+            const inDirectory = '/in';
+            let threw = false;
+            try {
+                await getWatermark(params, inDirectory);
+            } catch (e) {
+                assert.equal(e.message, 'Invalid or missing local file file/../../../../evilcode/elephant.jpg');
+                threw = true;
+            }
+            assert.ok(threw);
+        });
+
+        it('should fail because of missing localfile in worker test mode', async () => {
+            process.env.WORKER_TEST_MODE = true;
+            const params = {
+                watermarkContent: 'elephant.jpg'
+            };
+            const inDirectory = '/in';
+            let threw = false;
+            try {
+                await getWatermark(params, inDirectory);
+            } catch (e) {
+                assert.equal(e.message, 'Invalid or missing local file elephant.jpg');
+                threw = true;
+            }
+            assert.ok(threw);
+        });
+
+        it('should fail because watermark asset is not png', async () => {
+            let threw = false;
+            try {
+                await getWatermark({ watermarkContent: "https://example.com/photo/elephant.jpg" });
+            } catch (e) {
+                assert.equal(e.reason, 'RenditionFormatUnsupported');
```

Comment:
You should rather use assert.throws.

Reply:
I can't use assert.throw with async/await - I can either use try/catch or then/catch

Reply:
ack